clcache.py

```diff
@@ -313,26 +313,17 @@ def section(self, key):
     def sections(self):
         return (CompilerArtifactsSection(path) for path in childDirectories(self._compilerArtifactsRootDir))
 
-    def removeObjects(self, stats, removedObjects):
-        for o in removedObjects:
-            dirPath = self.section(o).cacheEntryDir(o)
-            if not os.path.exists(dirPath):
-                continue  # May be if object already evicted.
-            objectPath = os.path.join(dirPath, "object")
-            if os.path.exists(objectPath):
-                # May be absent if this if cached compiler
-                # output (for preprocess-only).
-                fileStat = os.stat(objectPath)
-                stats.unregisterCacheEntry(fileStat.st_size)
```

Comment:
I was thinking about making it private but then thought that there should be a public interface to remove an entry from CompilerArtifactsRepository. This is something that should be tested and can be reused later (i.e. some follow-up of 2.)

2.: This is about the list in `manifest.includesContentToObjectMap`, not the includes list. The idea to have a maximum number of entries in this list is not too bad in general. But the implementation was not good: Why pop the last item of a dict? Is this a random element? Is it a recent entry? Is it an old entry? 39765c24fd indeed removed this semi-clever magic, mostly because it (temporarily) removed the ability to change existing manifests. So right now `MAX_MANIFEST_HASHES` is unused. The question is, if this ever needs to be re-implemented, because it will only be used, when the same source file is compiled with more than 100 different include values. And even then there is no reason to stop at 100. Allowing 1000 entries is just fine as well.